server/server.go

```diff
@@ -24,18 +23,59 @@ import (
 const Version = "2.0.1b"
 
 type server struct {
-	client       *etcd.Client
+	getter Getter
+	config *Config
+
+	group        *sync.WaitGroup
 	dnsUDPclient *dns.Client // used for forwarding queries
 	dnsTCPclient *dns.Client // used for forwarding queries
-	config       *Config
-	group        *sync.WaitGroup
 	scache       *cache.Cache
 	rcache       *cache.Cache
 }
 
+type Getter interface {
+	Records(name string, exact bool) ([]msg.Service, error)
```

Comment:
[ Quoting <notifications@github.com> in "Re: [skydns] WIP - Abstract record ..." ]
>>  	scache       *cache.Cache
>>  	rcache       *cache.Cache
>>  }
>>
>> +type Getter interface {
>> +	Records(name string, exact bool) ([]msg.Service, error)
>
>Agree, I was only using your example :)
>
>Backend or RecordRetriever or RecordLoader or RecordStore all work for me.

Ok, let's go with Backend then.

/Miek

--
Miek Gieben